ultravox/training/train.py

```diff
@@ -179,29 +184,39 @@ def main() -> None:
                 mds_batch_size=args.batch_size,
             ),
         )
-        val_dataset = prepare_dataset(
-            dataset_names=args.data_sets,
-            train_on_inputs=args.train_on_inputs,
-            repeat_data=args.repeat_data,
-            processor=processor,
-            num_samples=args.val_num_samples,
-            data_args=datasets.VoiceDatasetArgs(
-                num_prompts=1,
-                data_dir=args.data_dir,
-                shuffle=False,
-                max_audio_duration_secs=16,
-                use_mds=args.mds,
-                mds_batch_size=args.batch_size,
-            ),
+        val_ds_args = datasets.VoiceDatasetArgs(
+            num_prompts=1,
+            data_dir=args.data_dir,
+            shuffle=False,
+            max_audio_duration_secs=16,
+            use_mds=args.mds,
+            mds_batch_size=args.batch_size,
         )
+        val_ds_args_text = copy.copy(val_ds_args)
+        val_ds_args_text.include_audio = False
+        val_datasets = {
+            k: prepare_dataset(
+                dataset_names=val_sets[k],
+                train_on_inputs=args.train_on_inputs,
+                repeat_data=args.repeat_data,
+                processor=processor,
+                num_samples=(
```

Comment:
I didn't follow what's happening with "matchtrain". Can you add a comment to explain?

Reply:
Will do. I was struggling with the name too. Let's disambiguate here first.

The idea is that classically both training and validation loss come from the same data distribution. Having that still has some advantages, simplest one being a way to show overfitting.

We have broken that assumption with the "dataset specific validations", but I still wanted to have the original. I wasn't sure how to name this. Since the point was to "match the training dataset distribution", I called it "matchtrain", but I'm open to ideas.

Reply:
Done.